src/Components/ProductFilters.jsx

```diff
@@ -6,20 +6,38 @@ function classNames(...classes) {
   return classes.filter(Boolean).join(" ");
 }
 
-export default function ProductFilters({ filterOptions, setFilterOptions, sortOptions, setSortOptions }) {
+export default function ProductFilters({
+  filterOptions,
+  setFilterOptions,
+  sortOptions,
+  setSortOptions,
+}) {
+  const sortProducts = (name) => {
+    let optionsArr = [...sortOptions];
+    optionsArr.map((item) => {
+      if (item.name === name) {
+        // switch on the sorting method clicked by user
+        item.current = true;
+      }
+      // switch off a previously selected sorting method if it had been selected
+      else item.current = false;
+    });
+    setSortOptions((prevState) => optionsArr);
+  };
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** I like how you separated the logic into a method, and simply call it below in the click handler. Good work!‏